standalone-metastore/metastore-common/src/main/thrift/hive_metastore.thrift

```diff
@@ -1249,6 +1249,24 @@ struct ShowCompactResponse {
     1: required list<ShowCompactResponseElement> compacts,
 }
 
+struct GetLatestCompactionInfoRequest {
+    1: required string dbname,
+    2: required string tablename,
+    3: optional list<string> partitionnames,
+}
+
+struct LatestCompactionInfo {
+    1: required i64 id,
+    2: optional string partitionname,
+    3: required CompactionType type,
+}
+
+struct GetLatestCompactionInfoResponse {
+    1: required string dbname,
+    2: required string tablename,
```

Comment:
Why do we need this? Do we anticipate filters in the `GetLatestCompactionInfoRequest`?

Reply:
@pvary I removed these two fields in response. This API is required by metadata cache server to get the latest compaction event for a table/partition so we need to filter by table/partition.